KubeArmor/core/unorchestratedUpdates.go

```diff
@@ -426,15 +426,70 @@ func (dm *KubeArmorDaemon) ParseAndUpdateContainerSecurityPolicy(event tp.K8sKub
 	appArmorAnnotations := map[string]string{}
 	appArmorAnnotations[containername] = "kubearmor_" + containername
 
+	i := -1
 	newPoint := tp.EndPoint{}
+	renamedContainers := []tp.RenamedContainer{}
 
-	i := -1
+	for idx, endPoint := range dm.EndPoints {
+		if kl.MatchIdentities(secPolicy.Spec.Selector.Identities, endPoint.Identities) && i < 0 {
+			i = idx
+			newPoint = endPoint
+		}
+
+		for policyIndex, policy := range endPoint.SecurityPolicies {
+			if policy.Metadata["namespaceName"] == secPolicy.Metadata["namespaceName"] && policy.Metadata["policyName"] == secPolicy.Metadata["policyName"] && endPoint.EndPointName != containername {
+				renamedContainer := tp.RenamedContainer{
+					EndPointName:       endPoint.EndPointName,
+					DeletedPolicyIndex: policyIndex,
+				}
+				renamedContainers = append(renamedContainers, renamedContainer)
```

Comment:
Any particular reason to store it in an array and loop over it again? Why not just call the deletion code here itself?